spec/controllers/admin/entry_templates_controller_spec.rb

```diff
@@ -4,6 +4,93 @@ module Admin
   describe EntryTemplatesController do
     render_views
 
+    describe '#new' do
+      it 'displays only themes that are registered for the entry type' do
+        pageflow_configure do |config|
+          ironed = Pageflow::TestEntryType.register(config, name: 'ironed')
+          steeled = Pageflow::TestEntryType.register(config, name: 'steeled')
+          config.for_entry_type(ironed) do |entry_type_config|
+            entry_type_config.themes.register(:iron_ore)
+          end
+          config.for_entry_type(steeled) do |entry_type_config|
+            entry_type_config.themes.register(:iron_ore)
+            entry_type_config.themes.register(:carbon)
+          end
+        end
+        admin = create(:user, :admin)
+        sign_in(admin, scope: :user)
+
+        get(:new,
+            params: {
+              account_id: admin.accounts.first.id,
+              entry_type_name: 'ironed'
+            })
+
+        expect(response.body).to have_text('iron_ore')
+        expect(response.body).not_to have_text('carbon')
```

Comment:
**thought**: Should one use a more specific selector here?

Since the names are such original, I see the advantage of just asserting "this shall not be mentioned". On the other hand I had to stop and think whether `have_text` actually matches text inside `<option>` tags. Using a more specific matcher like `have_selector('[name="entry_template[theme_name]"] option', text: 'iron_ore')` etc. would maybe make it clearer what we are actually looking for here.

Reply:
Indeed, the original names are meant to work around incidental matching. Matching this specific selector would be a coupling of the spec to Active Admin's current implementation, if there isn't some longer-lasting convention at work here that I'm not aware of.

Even if we just match for a select option with that text or whatever kind of input, we'd depend somewhat on implementation. We might at some point want to use some strange searchable select that doesn't use a select input under the hood, or maybe Active Admin starts using a seed json to populate form inputs, or something like that. I can still see how matching for a select option would make this spec more readable, and it might be unlikely that we switch this part of the implementation in a way that it doesn't use a select option. Also, it might happen that somebody manages to do something very strange like putting all available theme names into some other visible element.

Still, given Pageflow's long lifecycle, I remain in favor of just leaving this as it is. I have the feeling that this way, these specs might be less likely to break because of unrelated changes, and I think the risk of other strange changes that make these specs pass even though they shouldn't is negligible

Reply:
Yeah, you are probably right. Let's leave it like this.